pimd: stop KATs on one interface

anlancs opened the pull request:
Since #10569 is a workaround for #10782, this PR continues:
    
Immediately stop KATs on the interface if this interface is down or not
pim-enabled.

anlancs commented:
ci:rerun

mobash-rasool commented:
It seems the CI is failing due to your changes. rerun wont fix it. Please check.

anlancs committed:
pimd: stop KATs on one interface

Since #10569 is a workaround for #10782, this PR continues:

Immediately stop KATs on the interface if this interface is down or not
pim-enabled.

Signed-off-by: anlan_cs <vic.lan@pica8.com>

anlancs commented:
ci:rerun for ospf failed

anlancs commented:
ci:rerun for ospf failed again

anlancs commented:
ci:rerun why

anlancs commented:
@mobash-rasool Happy to see CI passed, please review it. Thanks.

mobash-rasool commented:
@anlancs : The issue which you were trying to fix here is closed(fixed) in another way. Can we close this PR?

anlancs commented:
@mobash-rasool I forgot details on it. But i have tested latest code just now, after `no ip pim` on interface of the upstream, the kat for it is still not deleted/stopped.

riw777 commented:
@frrbot autoclose in 1 month
